src/codeu/chat/common/User.java

```diff
@@ -48,6 +52,11 @@ public User read(InputStream in) throws IOException {
     }
   };
 
+  public Collection<Uuid> UserSet = new HashSet<Uuid>(64);
```

Comment:
You don't need both a User/Convo Set and Update map. The update maps can perform the same checks as the sets. Instead of HashSet.contains, you would just call HashMap.containsKey on the maps.